test/common-tests/acbs-authentication-api-tests.ts

```diff
@@ -0,0 +1,134 @@
+import { ENVIRONMENT_VARIABLES } from '@ukef-test/support/environment-variables';
+import { RandomValueGenerator } from '@ukef-test/support/generator/random-value-generator';
+import nock from 'nock';
+import request from 'supertest';
+
+interface AcbsAuthenticationErrorCasesTestOptions {
+  givenRequestWouldOtherwiseSucceed: () => void;
+  makeRequest: () => request.Test;
+}
+
+interface AcbsAuthenticationTestHooks {
+  idToken: string;
+  givenAuthenticationWithTheIdpSucceeds: () => void;
+}
+
+// Note that these tests assume the caller will use nock to abort pending requests
+// and clean all mocks after each test
```

Comment:
This function relies on the calling code to clean up `nock` correctly in an `afterEach` hook. We could move that into the function instead if we'd prefer.

For most use cases that I can imagine, it should be fine to move it into this function. The downside of that approach is that there are some edge cases where the calling code wouldn't want these tests to clean up `nock` themselves

We can always decide to change this later